File: src/main/java/ru/umeta/libraryintegration/util/ByteTo32SpreadAlgorithm.java
package ru.umeta.libraryintegration.util;

/**
 * Created by k.kosolapov on 12/8/2015.
 */
public class ByteTo32SpreadAlgorithm {
    public static int getHash(String value) {
        char[] chars = value.toCharArray();
        int result = 0;
        for (char character : chars) {
            byte b;
            result = (result << 8) + (byte) character;
        }
        //if the last bit is one shift it to the left on 16 bits.
        if ((result & 1) == 1) {
            result = result << 16;
        }
        return result;
    }
}

File: src/main/java/ru/umeta/libraryintegration/util/ByteTo32SpreadAlgorithm.kt
package ru.umeta.libraryintegration.util

import java.security.MessageDigest

/**
 * Created by k.kosolapov on 12/8/2015.
 */
object ByteTo32SpreadAlgorithm {
    fun getHash(value: String): Int {
        val chars = value.toCharArray()
        var result = 0
        for (character in chars) {
            result = (result shl 8) + character.toByte()
        }
        //if the last bit is one shift it to the left on 16 bits.
        if ((result and 1) == 1) {
            result = result shl 16
        }
        return result
    }
}
